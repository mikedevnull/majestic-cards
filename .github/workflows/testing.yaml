name: Test pipeline

on:
  push:
    branches: ["main"]
  pull_request:
    branches: ["main"]

jobs:
  build-test-and-lint:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: ["lts/*"]

    steps:
      - uses: actions/checkout@v4
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v4
        with:
          node-version: ${{ matrix.node-version }}
          cache: "npm"
      - run: npm ci
      - run: npm run build
      - run: npm run lint
      - run: npm run test
      - name: Install Playwright Browsers
        run: npx playwright install --with-deps
      - name: Run Playwright tests
        run: npm run test:e2e
      - uses: actions/upload-artifact@v4
        if: ${{ !cancelled() }}
        with:
          name: playwright-report
          path: playwright-report/
          retention-days: 2
