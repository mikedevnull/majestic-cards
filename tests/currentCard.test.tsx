import { createRemixStub } from "@remix-run/testing";
import { render, screen, waitFor } from "@testing-library/react";
import CurrentCardDisplay from "~/routes/_index";

test("Renders if there's currently no active card", async () => {
  const RemixStub = createRemixStub([
    {
      path: "/",
      Component: CurrentCardDisplay,
      loader() {
        return { activeCard: undefined };
      },
    },
  ]);

  render(<RemixStub />);

  await waitFor(() => screen.findByText("No active card in reader"));
});
