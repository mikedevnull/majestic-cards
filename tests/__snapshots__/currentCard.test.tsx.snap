// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders if there's currently no active card 1`] = `
<div>
  <div
    class="_card_2666b0 _cardInactive_2666b0"
  >
    <div
      class="_cardImage_2666b0"
    />
    <span>
      No active card in reader
    </span>
  </div>
</div>
`;

exports[`Renders placeholder text and image for unkown cards without data 1`] = `
<div>
  <div
    class="_card_2666b0 "
  >
    <div
      class="_cardImage_2666b0"
    >
      <img
        alt="No image"
        src="/app/assets/file-64.png"
      />
    </div>
    <span>
      Unknown card in reader
    </span>
  </div>
</div>
`;
